lib/index.js

```diff
@@ -477,47 +487,50 @@ Socket.prototype.send = function(msg, flags) {
 // This helper is called from `send` above, and in response to
 // the watcher noticing the signaller fd is readable.
 Socket.prototype._flush = function() {
-  var args;
+  var flags, args, emitArgs;
 
   // Don't allow recursive flush invocation as it can lead to stack
   // exhaustion and write starvation
   if (this._flushing) return;
 
-  if(zmq.STATE_CLOSED == this._zmq.state) return;
```

Comment:
Since we test for _zmq.state immediately when entering the while-loop, there's no need for it here.